helm/values.yaml

```diff
@@ -64,6 +64,11 @@ serviceMonitor:
   # -- ServiceMonitor scrape timeout
   # scrapeTimeout: 10s
 
+# ---------------------------------------------------------------------
+# -- Option to provide an existing secret with all the values from the config section
+# ---------------------------------------------------------------------
+existingSecret: {}
```

Comment:
I'd put `existingSecret` under the `config` section for better context, but I have no strong opinion here.

@allanger Do you mind checking the PR as well whenever you have some time? 🙂

Reply:
Also, I've added `helm-docs` to generate `README.md` for chart values, and it's sensitive to the comments format (using them for description). So if you could remove delimiters (line 67 and 69), that'd be cool.

Actually, if you could run `helm-docs --sort-values-order file` command (from [here](https://github.com/norwoodj/helm-docs)) to regenerate `README.md`, that'd be amazing, so we could have a smooth release.

I'll add the notes to the contribution guide later. 🙂👍

Reply:
@burningalchemist I'll check tomorrow

Reply:
I also think that config related values should be under the config key. But the problem then is that currently we add the whole `config` object to the secret, I'd suggest to do something like this

```yaml
config:
  existingSecret: SECRETNAME
  inline: 
    global: ...
    ...
```

And then, if `Values.config.existingSecret` is not empty, we mount the existing one, otherwise, we put `Values.config.inline` to the templated secret

Reply:
And it also would mean the major version bump, because it's a breaking change

Reply:
I'm not concerned about breaking changes since we're under `0.x` (zerover) so far, so breaking changes are intended, expected, but also should be announced. So we'd bump up the minor as usual.

I don't really like `inline` (although I understand the intention), let me think about it. 😉

Reply:
Also we can do it like 
```yaml
config:
  from:
    kind:
    name:
    key:
  value: |
    ...
```

like here https://github.com/burningalchemist/sql_exporter/pull/399#issuecomment-1819284076

Reply:
Oh, this one looks good. 👍

Reply:
I'm not sure that `value: | ` is a good idea, I guess it's better to handle `value` as an object in this case, and cast it to yaml in the template already, then users will be able to replace parts of it.

I'd say that it should be 
```yaml
config:
  from:
    kind:
    name:
    key:
  value: 
    global: ...
    ...
```

And then in the template it still can be `{{ toYaml .Values.config.value }}`

Reply:
Fair enough, I was mostly looking at the structure. But yeah, agreed. 👍